For regular strings avoid igbinary serialization overhead

iliaal opened the pull request:
This simple change ensures that string data is stored as is without introducing overhead in speed/memory involved in serialization/deserialization steps.

In benchmarks this shows 1-2% speed improvement in terms of speed, but more critically for shorter strings it eliminates added network overhead resulting from igbinary header.

iliaal committed:
fixed checks

iliaal committed:
fixed checks

iliaal committed:
Fixed check

iliaal committed:
refactor based on develop branch

yatsukhnenko commented:
I don't like changes like this. It adds strange unexpected behaviour in exchange for 1-2% performance. Users always may disable serialization via `setOption` before sending any data not only strings and all controll and responsibility are on their side.

iliaal commented:
> I don't like changes like this. It adds strange unexpected behaviour in exchange for 1-2% performance. Users always may disable serialization via `setOption` before sending any data not only strings and all controll and responsibility are on their side.

I appreciate the feedback, I made it into configuration option specifically so that it can be controlled by the user and turned off/on and is disabled by default. This makes it easier then sprinkling code with setOption() to change value and restore it back, when using a mixed environment where complex values (Arrays & Objects) should be serialized and strings left as they are.

The change from logic perspective as far as I can tell is very low risk and adds minimal (if any) cognitive overhead to internal extension code logic.

I guess another approach would be to do something like a smart_igbinary serializer that works like igbinary but implements different logic for scalar values and is distinct serialization logic within code flow. If that would be more agreeable I am happy to adjust the patch.

yatsukhnenko commented:
IMO there should be a division of responsibility. For phpredis all serializers look like a black box that takes a value at the input and gives a string at the output. How it works and how it can be optimized needs to be changed at the level of the serializer code. Something like `smart_igbinary` may be used but it should be external library (because I don't want to answer the questions about its behavior :smile:) May be @michael-grunder thinks different

iliaal committed:
redis clone

iliaal committed:
Create new socket on clone

iliaal committed:
Merge branch 'develop' of https://github.com/phpredis/phpredis into redis_clone_develop

iliaal committed:
Added auth support

Jan-E committed:
PHP 8 compatibility Windows

yatsukhnenko committed:
5.3.2

defender-11 committed:
Fixed#1895

iliaal committed:
Merge branch 'develop' into redis_clone_develop

michael-grunder committed:
Prepare 5.3.3 release

michael-grunder committed:
Disable clone handlers for Redis and RedisCluster.

Presently PhpRedis will segfault if users attempt to clone either the
Redis or RedisCluster objects.

We plan on investigating proper clone support in future releases, but
for now just diable the handlers.

michael-grunder committed:
Update changelog for 5.3.3

michael-grunder committed:
5.3.2

michael-grunder committed:
Normalize Redis callback prototypes and stop typecasting. (#1935)

michael-grunder committed:
Fix ZSTD decompression on bad data.

ZSTD uses two defined error numbers to inform the caller when the
compressed data is invalid (e.g. wrong magic number) or the size is
unknown.

We should always know the size so abort if ZSTD returns either to us.

Fixes: #1936

aolley committed:
Pass compression flag when performing HMGET (#1945)

Without this, performing a HMGET call fails to decompress the data before
returning it to php.

michael-grunder committed:
Prepare for 5.3.4 release

michael-grunder committed:
Fix PhpRedis session tests to soften timing issues

Rework the session locking unit tests to be less reliant on arbitrary
sleep calls which can be very troublesome when running in Travis and
especially when running in Travis under valgrind.

michael-grunder committed:
Prepare for 5.3.4 v2

michael-grunder committed:
More Travis-CI fixes

michael-grunder committed:
5.3.4 - Move the date back one day

iliaal committed:
Merge branch 'develop' of github.com:phpredis/phpredis into redis_clone_develop

iliaal committed:
Merge branch 'redis_clone_develop' into develop_simple_strings

iliaal committed:
fixes

iliaal committed:
merge

iliaal committed:
Merge branch 'release/5.3.4' of github.com:phpredis/phpredis into develop_simple_strings

iliaal committed:
1st patch

iliaal committed:
merge

iliaal committed:
Added checks

iliaal committed:
Added lz4 test

iliaal committed:
removed redundant code
